### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Satont**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -27,19 +27,16 @@ export class RedisAdapter<T> implements StorageAdapter<T> {
...
-      const value = await this.redis.hmget(key) as unknown as T;
-      //console.log('v', value)
-      return value;
-    } catch (e) {
-      console.error(e);
-      return null;
+    if (!await this.redis.get(key)) {
+      return undefined;
     }
+    const value = await this.redis.get(key) as unknown as string;
```
Comment (KnorpelSenf) [inline src/index.ts:33]: You're reading from Redis twice. It's better to query it only once, store the result in a variable, and operate on the same value.

File: src/index.ts [hunk 2]
```diff
@@ -27,19 +27,16 @@ export class RedisAdapter<T> implements StorageAdapter<T> {
...
+    return JSON.parse(value) as unknown as T;
   }
 
   async write(key: string, value: T) {
     try {
-      await this.redis.hmset(key, value as any);
+      await this.redis.set(key, JSON.stringify(value));
     } catch (e) {
       console.error(e);
     }
```
Comment (KnorpelSenf) [inline src/index.ts:42]: Catching errors should ideally be left to user code. That way, users can decide themselves if they want to `console.log` the error, or do something else with it.

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]
LGTM generally. Just some minor changes, see below.

File: src/index.ts (hunk 1 above)
Comment (Satont) [reply src/index.ts]: Fixed.

File: src/index.ts (hunk 2 above)
Comment (Satont) [reply src/index.ts:42]: Yea i wasn't sure should i leave that or make to user job, thank you.

File: src/index.ts [hunk 3]
```diff
@@ -27,29 +26,18 @@ export class RedisAdapter<T> implements StorageAdapter<T> {
...
   async read(key: string) {
-    try {
-      const value = await this.redis.hmget(key) as unknown as T;
-      //console.log('v', value)
-      return value;
-    } catch (e) {
-      console.error(e);
-      return null;
+    const session = await this.redis.get(key);
+    if (!session) {
```
Comment (KnorpelSenf) [inline src/index.ts:30]: Fails for `T extends string` when storing empty strings. Use `!= null` to check for nullish rather than falsy values.

Comment (KnorpelSenf) [reply src/index.ts:30]: The same is true if `T` is of type `number` and a user stores `0` as session data.

**Pull request merged**